co2calculator/distances.py

```diff
@@ -542,11 +602,15 @@ def get_distance(request: DistanceRequest) -> Kilometer:
         return _apply_detour(distance, request.transportation_mode)
 
     if request.transportation_mode == TransportationMode.FERRY:
-        # todo: Do we have a way of checking if there even exists a ferry connection between the given cities (or if the
-        #  cities even have a port?
         _, _, geom_start, _ = geocoding_structured(request.start.dict())
         _, _, geom_dest, _ = geocoding_structured(request.destination.dict())
-        # compute great circle distance between ports
-        distance = haversine(geom_start[1], geom_start[0], geom_dest[1], geom_dest[0])
 
-        return _apply_detour(distance, request.transportation_mode)
+        # hardcoding not ideal, profile should be determined based on specified "seating type"
+        distance, distance_total = get_route_ferry(
+            [geom_start, geom_dest], profile="foot-walking"
+        )
+
+        # if "seating" is "Car passenger", the remaining distance should be calculated as car trip ...
+        # TODO: implement this
+        remaining_distance = distance - distance_total
```

Comment:
@redfrexx @veitu Should we implement this?
Ideally, there should not be too much remaining distance and the users should enter car sections of the trip themselves.. it would be a cool feature, but not too straightforward to implement.. :D

Reply:
I think it would be nice to have but it's not necessary in my opinion. I think ferry trips as such are already quite a special case and for now we can keep it this way that drivers enter separate care trips for the car sections.